mitodl/mitxpro
# feat(catalog): add course topics filter

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxpro/issues/2591

#### What's this PR do?

- Adds a course topic filter on the catalog page.
- Updates banner and featured product position.

#### How should this be manually tested?

- Verify the designs
- Verify that the filters are working fine
  - Visit wagtail and create some parent course topics having parent=null and some child topics with parents.
  - Visit wagtail admin, Visit course pages, and Add some topics.
  - Parent topics should work fine, If a child topic is added to a course it should appear when the parent filter is applied.
  - Same should be verified for programs, A program having courses with the filtered title should appear in the catalog.

#### Screenshots (if appropriate)
Laptop 1440:
<img width="1753" alt="Screen Shot 2023-03-22 at 5 55 29 PM" src="https://user-images.githubusercontent.com/52656433/226911259-436c3e19-3ba3-4d85-b788-4775e87fed21.png">


Laptop 1024:
<img width="1026" alt="Screen Shot 2023-03-22 at 5 55 57 PM" src="https://user-images.githubusercontent.com/52656433/226911378-fe4faad6-a0ad-47fa-a334-e4450b23c97b.png">


Tablet:
<img width="789" alt="Screen Shot 2023-03-22 at 5 56 12 PM" src="https://user-images.githubusercontent.com/52656433/226911512-8bb2e948-d000-48aa-b120-a97a6039a3da.png">


Mobile L:
<img width="448" alt="Screen Shot 2023-03-22 at 5 56 21 PM" src="https://user-images.githubusercontent.com/52656433/226911568-82d018f5-83c9-4019-b99c-ca1bb84ad946.png">
